engine/src/main/java/org/terasology/engine/audio/openAL/staticSound/OpenALSound.java

```diff
@@ -119,18 +119,25 @@ protected void doReload(StaticSoundData newData) {
         }
     }
 
-    private static class DisposalAction implements Runnable {
+    public static class DisposalAction implements DisposableResource {
         private final ResourceUrn urn;
         private int bufferId;
-        private final WeakReference<OpenALSound> asset;
+        private WeakReference<OpenALSound> asset;
 
-        DisposalAction(ResourceUrn urn, OpenALSound openALSound) {
+        public DisposalAction(ResourceUrn urn) {
```

Comment:
why is this exposed, shouldn't this just be related to sounds resources. maybe protected?